### PR TITLE
allegro-internal/flex-roadmap#804 Added/removed logs

### PR DESCRIPTION
**Pull request by nastassia-dailidava**

### CODE REVIEW

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java [hunk 1]
```diff
@@ -268,12 +268,15 @@ public DeltaWatch createDeltaWatch(
...
+            LOGGER.debug("removing watch {}", watchId);
+            status.removeWatch(watchId);
+        });
+
         if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("open watch {} for {}[{}] from node {} for version {}",
+            LOGGER.debug("open watch {} for {} from node {} for version {}",
                 watchId,
                 url,
-                String.join(", ", resources),
```
Comment (kozjan) [inline envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:276]: why is it getting removed?

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java [hunk 2]
```diff
@@ -371,12 +374,6 @@ protected void respondWithSpecificOrder(T group,
                     String version = snapshot.version(watch.request().getResourceType(), watch.request().getResourceNamesList());
 
                     if (!watch.request().getVersionInfo().equals(version)) {
-                        if (LOGGER.isDebugEnabled()) {
-                            LOGGER.debug("responding to open watch {}[{}] with new version {}",
-                                id,
-                                String.join(", ", watch.request().getResourceNamesList()),
-                                version);
-                        }
```
Comment (kozjan) [inline envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:379]: why is it getting removed?

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java [hunk 3]
```diff
@@ -413,12 +410,6 @@ protected void respondWithSpecificOrder(T group,
                     String version = snapshot.version(watch.request().getResourceType(), Collections.emptyList());
 
                     if (!watch.version().equals(version)) {
-                        if (LOGGER.isDebugEnabled()) {
-                            LOGGER.debug("responding to open watch {}[{}] with new version {}",
-                                id,
-                                String.join(", ", watch.trackedResources().keySet()),
-                                version);
-                        }
```
Comment (kozjan) [inline envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:421]: same here, why?

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java [hunk 4]
```diff
@@ -268,12 +268,15 @@ public DeltaWatch createDeltaWatch(
                                                            String version) {
         long watchId = watchCount.incrementAndGet();
         status.setWatch(watchId, watch);
-        watch.setStop(() -> status.removeWatch(watchId));
+        watch.setStop(() -> {
+            LOGGER.debug("removing watch {}", watchId);
```
Comment (kozjan) [inline envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:272]: how will this help you with investigating the issue of removing snapshots with existing watches?

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java (hunk 2 above)
Comment (nastassia-dailidava) [reply envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:379]: it produces enormous amount of logs while not adding any useful information and it actually makes it much more difficult to debug

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java (hunk 3 above)
Comment (nastassia-dailidava) [reply envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:421]: responded above

File: envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java (hunk 4 above)
Comment (nastassia-dailidava) [reply envoy-control-core/src/main/java/pl/allegro/tech/servicemesh/envoycontrol/SimpleCache.java:272]: snapshot is getting cleared when watch is removed and id of watch can be connected to the event when it was created while that one can be connected to events when instances of a services were changed, not sure if that will be enough, but right now there's literally nothing that can help debug it

**Review by kozjan — APPROVED** [review]

**Pull request merged**